mods/explorer-ctrlq-newfile.wh.cpp

```diff
@@ -0,0 +1,463 @@
+        std::wstring ret = p;
+        CoTaskMemFree(p);
+        return ret;
+    }
+    wchar_t buf[MAX_PATH] = {0};
+    if (SUCCEEDED(SHGetFolderPathW(nullptr, CSIDL_DESKTOPDIRECTORY, nullptr, SHGFP_TYPE_CURRENT, buf)))
+        return buf;
+    return L"";
+}
+
+static std::wstring MakeUniqueFolderName(const std::wstring& dir) {
+    auto join = [](const std::wstring& a, const std::wstring& b) {
+        if (a.empty()) return b;
+        if (a.back() == L'\\' || a.back() == L'/') return a + b;
+        return a + L'\\' + b;
+    };
+    
+    std::wstring base = L"New folder";
+    std::wstring path = join(dir, base);
+    if (!PathFileExistsW(path.c_str())) return path;
+    
+    for (int i = 2; i < 10000; ++i) {
+        wchar_t buf[64];
+        swprintf(buf, ARRAYSIZE(buf), L"New folder (%d)", i);
+        path = join(dir, buf);
+        if (!PathFileExistsW(path.c_str())) return path;
+    }
+    return join(dir, L"New folder_new");
+}
+
+static bool CreateNewDirectory(const std::wstring& fullPath) {
+    if (CreateDirectoryW(fullPath.c_str(), nullptr)) {
+        return true;
+    }
+    return false;
+}
+
+static bool BindParentAndChildPIDL(const std::wstring& fullPath, IShellFolder** ppFolder,
+                                   PCUITEMID_CHILD* ppChild) {
+    if (!ppFolder || !ppChild) return false;
+    *ppFolder = nullptr;
+    *ppChild = nullptr;
+
+    PIDLIST_ABSOLUTE pidl = nullptr;
+    SFGAOF sf = 0;
+    if (FAILED(SHParseDisplayName(fullPath.c_str(), nullptr, &pidl, 0, &sf)) || !pidl) return false;
+
+    IShellFolder* pFolder = nullptr;
+    PCUITEMID_CHILD pChild = nullptr;
+    HRESULT hr = SHBindToParent(pidl, IID_PPV_ARGS(&pFolder), &pChild);
+    CoTaskMemFree(pidl);
+    if (FAILED(hr)) return false;
+
+    *ppFolder = pFolder;  // keep ref
+    *ppChild = pChild;    // pointer owned by parent bind
+    return true;
+}
+
+static void SelectAndRenameInWindow(HWND hwndExplorer, const std::wstring& fullPath) {
+    IWebBrowser2* wb = nullptr;
+    if (!GetExplorerBrowserByHWND(hwndExplorer, &wb) || !wb) return;
+
+    wchar_t folderPath[MAX_PATH * 2] = {0};
+    wcscpy_s(folderPath, fullPath.c_str());
+    PathRemoveFileSpecW(folderPath);
+
+    IServiceProvider* sp = nullptr;
+    if (FAILED(wb->QueryInterface(IID_PPV_ARGS(&sp))) || !sp) {
+        wb->Release();
+        return;
+    }
+
+    IShellBrowser* psb = nullptr;
+    HRESULT hr = sp->QueryService(SID_STopLevelBrowser, IID_PPV_ARGS(&psb));
+    sp->Release();
+    if (FAILED(hr) || !psb) {
+        wb->Release();
+        return;
+    }
+
+    IShellView* psv = nullptr;
+    hr = psb->QueryActiveShellView(&psv);
+    psb->Release();
+    if (FAILED(hr) || !psv) {
+        wb->Release();
+        return;
+    }
+
+    IShellFolder* pParent = nullptr;
+    PCUITEMID_CHILD pChild = nullptr;
+    if (BindParentAndChildPIDL(fullPath, &pParent, &pChild)) {
+        // This is where it was crashing - added safety
```

Comment:
Why does it crash? It's preferable to have a proper fix rather than using `Sleep` which just reduces the chances for a crash.